src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,478 @@
+
+    Args:
+        sif_cache_dir: Path to attempt locating/download SIF container image.
+        datasets_version: URL of singularity container (custom 'datasets' version if desired)
+
+    Returns:
+        `spython.main.client` instance of singularity container image housing 'datasets'.
+    """
+
+    # Set singularity cache dir from user defined path or use environment
+    if sif_cache_dir and sif_cache_dir.is_dir():
+        image_dl_path = Path(sif_cache_dir)
+        logging.info(f"Using user-defined cache_dir: '{image_dl_path}'")
+    elif os.environ.get("NXF_SINGULARITY_CACHEDIR"):
+        image_dl_path = Path(os.environ["NXF_SINGULARITY_CACHEDIR"])
+        logging.info(
+            f"Using preferred nextflow singularity cache dir 'NXF_SINGULARITY_CACHEDIR': {image_dl_path}"
+        )
+    elif os.environ.get("SINGULARITY_CACHEDIR"):
+        image_dl_path = Path(os.environ["SINGULARITY_CACHEDIR"])
+        logging.info(
+            f"Using the default singularity installation cache dir 'SINGULARITY_CACHEDIR': {image_dl_path}"
+        )
+    else:
+        image_dl_path = Path()
+        logging.warning(f"Unable to set singularity cache dir properly, using CWD {image_dl_path}")
+
+    # Set the datasets version URL
+    if datasets_version is None:
+        container_url = DATASETS_SINGULARITY["datasets_version_url"]
+        logging.info(f"Using default 'ncbi datasets' version '{container_url}'")
+    else:
+        container_url = datasets_version
+        logging.info(f"Using user defined 'ncbi datasets' version '{container_url}'")
+
+    # Pull or load pre-existing 'datasets' singularity container image.
+    datasets_image = Client.pull(container_url, stream=False, pull_folder=image_dl_path, quiet=True)
+
+    return datasets_image
+
+
+def check_parameterization(input_cores: Path, input_accessions: Path, db_host: str, db_port: int) -> Path:
+    """Detect the kind of user input (cores/accessions) and determine any missing or
+    incorrect parameterization.
+
+    Args:
+        input_cores: input core(s) list file name.
+        input_accessions: input accession (s) list file name.
+        db_host: host server name
+        db_port: host serer port
+
+    Returns:
+        User input file used in assembly status querying
+    """
+    user_query_file: Path
+
+    # Input core names centered run
+    if input_cores:
+        user_query_file = input_cores
+        logging.info(f"Performing assembly status report using core db list file: {user_query_file}")
+        if db_host is None or db_port is None:
+            logging.critical(
+                "User must specify both arguments '--host' and '--port' when providing core database names."
+            )
+            sys.exit(1)
+    # Accession centered run
+    else:
+        user_query_file = input_accessions
+        logging.info(f"Performing assembly status report using INSDC accession list file: {user_query_file}")
+
+    return user_query_file
```

Comment:
Whilst the code is correct, I would simplify this function:
```suggestion
    # Input core names centered run
    if input_cores:
        logging.info(f"Performing assembly status report using core db list file: {input_cores}")
        if db_host is None or db_port is None:
            raise RuntimeError("Core database names require both arguments '--host' and '--port'")
        return input_cores
    # Accession centered run
    logging.info(f"Performing assembly status report using INSDC accession list file: {input_accessions}")
    return input_accessions
```